syne_tune/optimizer/schedulers/searchers/bayesopt/sklearn/contributed_surrogate_searcher.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import logging
+from typing import Optional, List, Dict, Any
+
+from syne_tune.optimizer.schedulers.searchers.bayesopt.models.meanstd_acqfunc_impl import (
+    EIAcquisitionFunction,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.models.sklearn_estimator import (
+    SklearnEstimatorWrapper,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.sklearn.estimator import (
+    SklearnEstimator,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.tuning_algorithms.base_classes import (
+    ScoringClassAndArgs,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.tuning_algorithms.defaults import (
+    DEFAULT_NUM_INITIAL_CANDIDATES,
+    DEFAULT_NUM_INITIAL_RANDOM_EVALUATIONS,
+)
+from syne_tune.optimizer.schedulers.searchers.gp_searcher_utils import (
+    decode_state,
+)
+from syne_tune.optimizer.schedulers.searchers.model_based_searcher import (
+    BayesianOptimizationSearcher,
+)
+from syne_tune.optimizer.schedulers.searchers.utils import make_hyperparameter_ranges
+
+logger = logging.getLogger(__name__)
+
+
+class ContributedSurrogateSearcher(BayesianOptimizationSearcher):
+    """Contributed Surrogate Bayesian optimization for FIFO scheduler
+
+    This searcher must be used with
+    :class:`~syne_tune.optimizer.schedulers.FIFOScheduler`. It provides
+    Bayesian optimization, based on a Contributed surrogate model.
+
+    Most of the implementation is generic in
+    :class:`~syne_tune.optimizer.schedulers.searchers.model_based_searcher.BayesianOptimizationSearcher`.
+    """
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        metric: str,
+        estimator: SklearnEstimator,
+        scoring_class_and_args: ScoringClassAndArgs = None,
```

Comment:
This should be `acquisition_class`.

To explain this. In GP-BO, you can use the AF for (1) scoring initial cands, and (2) fine-tune by local search. But for (1), you can also use something else. For example, ITS is *independent* of the AF.

Reply:
Sure we need `ScoringClassAndArgs`? I'd just use `AcquisitionClassAndArgs`.

Reply:
I have added a the `ScoringClassAndArgs` in `syne_tune/optimizer/schedulers/searchers/bayesopt/tuning_algorithms/base_classes.py`. Overall I think the skleanr searcher will rely on acquisition functions w.o gradients. The local optimizatin is already being skipped in https://github.com/awslabs/syne-tune/pull/684/files#diff-63315ef366b2e2b3cde9ea92943d6c9e8db65fcd630e8f4fce379f754fdc406eR99 so I dont think the gradients would be useful.

Note that `AcquisitionFunction` inherits from`ScoringFunction` so any acquitision function can be used here. See the launch example that uss EI.